bundle/regal/rules/bugs/sprintf_argument_mismatch.rego

```diff
@@ -0,0 +1,48 @@
+# METADATA
+# description: Mismatch in `sprintf` arguments count
+package regal.rules.bugs["sprintf-arguments-mismatch"]
+
+import rego.v1
+
+import data.regal.ast
+import data.regal.config
+import data.regal.result
+
+# METADATA
+# description: Missing capability for built-in `sprintf`
+# custom:
+#   severity: none
+notices contains result.notice(rego.metadata.chain()) if not "sprintf" in object.keys(config.capabilities.builtins)
+
+# METADATA
+# description: |
+#   Count the number of distinct arguments ("verbs", denoted by %) in the string argument,
+#   compare it to the number of items in the array (if known), and flag when the numbers
+#   don't match
+report contains violation if {
+	some fn
+	ast.function_calls[_][fn].name == "sprintf"
```

Comment:
This looks expensive. Have you ever tried using an reverse-index here? Like "all locations where function `sprintf` is used"

Reply:
I think we've mostly paid the cost at this point, as `ast.function_calls` essentially filters `ast.found.refs` keeping only function calls, and then providing them in compact format nicer to work with.

But yeah, `ast.found.refs` is a beast, and as I'm sure you remember, I tried to collect more types in that traversal, and things... broke :P